pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -225,6 +226,8 @@ private void buildBroadcastHashTable()
           // Just fill up the buffer.
           int remainingRows = _maxRowsInHashTable - _currentRowsInHashTable;
           container = container.subList(0, remainingRows);
+          OperatorStats operatorStats = _opChainStats.getOperatorStats(_context, _operatorId);
```

Comment:
Since we set it here, we can change `_resourceLimitExceededException` into local variable and only create it in `THROW` mode